src/github_runner_manager/github_client.py

```diff
@@ -258,3 +240,51 @@ def get_job_info(self, path: GitHubRepo, workflow_run_id: str, runner_name: str)
             ) from exc
 
         raise JobNotFoundError(f"Could not find job for runner {runner_name}.")
+
+    @catch_http_errors
+    def get_job_info(self, path: GitHubRepo, job_id: str) -> JobInfo:
+        """Get information about a job identified by the job id.
+
+        Args:
+            path: GitHub repository path in the format '<owner>/<repo>'.
+            job_id: The job id.
+
+        Returns:
+            The JSON response from the API.
+        """
+        job_raw = self._client.actions.get_job_for_workflow_run(
+            owner=path.owner,
+            repo=path.repo,
+            job_id=job_id,
+        )
+        return self._to_job_info(job_raw)
+
+    @staticmethod
+    def _to_job_info(job: dict) -> JobInfo:
+        """Convert the job dict to JobInfo.
+
+        Args:
+            job: The job dict.
+
+        Returns:
+            The JobInfo object.
+        """
+        # datetime strings should be in ISO 8601 format,
+        # but they can also use Z instead of
+        # +00:00, which is not supported by datetime.fromisoformat
+        created_at = datetime.fromisoformat(job["created_at"].replace("Z", "+00:00"))
+        started_at = datetime.fromisoformat(job["started_at"].replace("Z", "+00:00"))
+        # conclusion could be null per api schema, so we need to handle that
+        # though we would assume that it should always be present,
+        # as the job should be finished
```

Comment:
Same as above

Reply:
Thank you very much. As this is a fairly minor issue, and there's a follow-on PR planned fairly soon, I'll make a note of it and address it in a follow-on PR, to reduce the overhead of re-running tests and re-asking for approval for this PR.